can/message.py

```diff
@@ -65,7 +65,7 @@ def __str__(self):
 
         field_strings.append("DLC: {0:d}".format(self.dlc))
         data_strings = []
-        if self.data is not None:
+        if self.data is not None and len(self.data) > 0:
             for index in range(0, self.dlc):
```

Comment:
Yes. I didn't think enough before sending this PR. It'd make more sense to change "range(0, self.dlc)" to "range(0, len(self.data))". Is it ok if I close this PR and open another?